doc/user/migrating-existing-apis.md

```diff
@@ -0,0 +1,382 @@
+                type: string
+          ...
+    ```
+
+    The API server will validate each version by its own `schema` if the global `spec.validation` block is removed. No validation will be performed if a `schema` does not exist for a version and `spec.validation` does not exist.
+
+    If the CRD targets a Kubernetes 1.13+ cluster with the `CustomResourceWebhookConversion` feature enabled, converting between multiple versions can be done using a [conversion webhooks][crd-conv-webhook].
+
+    **Note:** read the [CRD versioning][crd-versions] docs for detailed CRD information, notes on conversion webhooks, and CRD versioning case studies.
+
+1. *Optional:* `spec.versions` elements have a `subresources` field that holds CR subresource information to override the global `spec.subresources` block. `spec.subresources` will be used by the API server to assess subresource requirements of any version in `spec.versions` that does not have a `subresources` block. If all versions have the same requirements, leave `spec.subresources` as-is and skip to the [following section](#golang-api-migrations-types-and-commonalities). If CRD versions differ in subresource requirements, add a `subresources` section in each `spec.versions` entry with differing requirements and add each subresource's spec and status as needed:
+
+    ```yaml
+    spec:
+      ...
+      version: v2
+      versions:
+      - name: v2
+        served: true
+        storage: true
+        subresources:
+          ...
+      - name: v1
+        served: true
+        storage: false
+        subresources:
+          ...
+    ```
+
+    Remove the global `spec.subresources` block if all versions have different subresource requirements.
+
+1. *Optional:* remove `spec.version`, as it is deprecated in favor of `spec.versions`.
+
+### Golang API Migrations: Types and Commonalities
+
+This version upgrade walkthrough demonstrates only one of several possible migration scenarios:
+
+- Group migration, ex. moving an API from group `operators.example.com/v1` to `new-group.example.com/v1alpha1`.
+- Kind change, ex. `CatalogSourceConfig` to `CatalogSourceConfigurer`.
+- Some combination of group, version, and kind migration.
+
+Each case is different; one may require many more changes than others. However there are several themes common to all:
+
+1. Using `operator-sdk add api` to create the necessary directory structure and files used in migration.
+    - Group migration using the same version, for each kind in the old group `operators.example.com` you want to migrate:
+
+      ```console
+      $ operator-sdk add api --api-version new-group.example.com/v1 --kind YourKind
+      ```
+
+    - Kind migration, using the same group and version as `CatalogSourceConfig`:
+
+      ```console
+      $ operator-sdk add api --api-version operators.example.com/v1 --kind CatalogSourceConfigurer
+      ```
+
+1. Moving code from one Go package to another, ex. from `v1` to `v2` and `shared`.
```

Comment:
```suggestion
2. Moving code from one Go package to another, ex. from `v1` to `v2` and `shared`.
```
Also fix the numbering below.

Reply:
GitHub markdown treats all numbers as sequentially increasing if they're in list format. The above renders as `2.`.

Reply:
Ah, TIL.
It's just jarring to look at in raw markdown at first, but if it works it's fine.

Reply:
This list format is easier with Git documents. You do not have to change list numbers when inserting/deleting list elements -> less lines changed in a commit.

Reply:
Agreed